averbis/core/_rest_client.py

```diff
@@ -2566,3 +2709,58 @@ def _create_zip_io(source: Union[IO, Path, str], path_in_zip: Union[Path, str] =
 
         zip_archive.seek(0)
         return zip_archive
+
+    @staticmethod
+    def _fetch_text_analysis_result_filename(source: Union[Path, IO, str]) -> str:
+        if isinstance(source, Path):
+            return Path(source).name
+
+        if isinstance(source, IOBase) and hasattr(source, "name"):
+            return source.name
+
+        if isinstance(source, Cas):
+            return "annotated_document.xmi"
+
+        raise ValueError("Unsupported source type - valid is [Path, IO, CAS]")
+
+    def _add_text_analysis_result_to_document(self, project_name, process_name, document_collection_name, source,
+                                              document_name, mime_type, typesystem):
+        filename = self._fetch_text_analysis_result_filename(source)
```

Comment:
Caller should be able to set the filename as a parameter, in particular for CAS uploads - but an override should also be possible for uploads from an IO object or path.